File: University/AdadeKhodeMaglubS1.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {

            String number = scanner.nextLine();
            String reverse = "";

            for(int i = 0; i < number.length(); i++) {
                reverse = reverse.concat(String.valueOf(number.charAt(number.length() - (i+1))));
            }

            if (reverse.equals(number))
                System.out.print("YES");
            else
                System.out.print("NO");

        }
    }
}

// </Ctrl+Z>

File: University/AdadeKhodeMaglubS1.rb
class Main
    def self.main()
        try ()
        {
            number = gets.chomp
            reverse = ""
            i = 0
            while (i < number.length)
                reverse = reverse.concat(number[number.length - (i + 1)].join(""))
                i += 1
            end
            if (reverse===number)
                print("YES")
            else
                print("NO")
            end
        }
    end
end
Main.main()
